packages/keystone/tasks.js

```diff
@@ -252,53 +337,99 @@ async function createWorker (keystoneModule) {
-        logger.info({ taskId: job.id, status: 'processing', task: getTaskLoggingContext(job) })
+    // Reapply queues configuration with worker startup config
+    if (get(config, '0', []).length > 0) {
+        let parsedConfig
         try {
-            return await executeTask(job.name, job.data.args, job)
-        } catch (error) {
-            logger.error({ taskId: job.id, status: 'error', error, task: getTaskLoggingContext(job) })
-            throw error
+            parsedConfig = JSON.parse(config[0])
+        } catch (e) {
+            throw new Error('Can\'t parse worker config. Please provide correct value')
         }
-    })
 
-    taskQueue.on('failed', function (job) {
-        logger.info({ taskId: job.id, status: 'failed', task: getTaskLoggingContext(job) })
-    })
+        if (parsedConfig['include'] && parsedConfig['include'].length > 0) {
+            const queuesToDelete = Array.from(QUEUES.entries()).filter(queue => !parsedConfig['include'].includes(queue[0]))
+
+            for (const [queueName] of queuesToDelete) {
+                QUEUES.delete(queueName)
+            }
+        }
+
+        if (parsedConfig['exclude'] && parsedConfig['exclude'].length > 0) {
+            const queuesToDelete = Array.from(QUEUES.entries()).filter(queue => parsedConfig['exclude'].includes(queue[0]))
+            for (const [queueName] of queuesToDelete) {
+                await removeQueue(queueName)
+            }
+        }
+    }
 
-    taskQueue.on('completed', function (job) {
-        logger.info({ taskId: job.id, status: 'completed', task: getTaskLoggingContext(job), t0: job.finishedOn - job.timestamp, t1: job.processedOn - job.timestamp, t2: job.finishedOn - job.processedOn })
-        gauge({ name: `worker.${job.name}ExecutionTime`, value: job.finishedOn - job.processedOn })
+    const activeQueues = Array.from(QUEUES.entries())
+
+    // Apply callbacks to each created queue
+    activeQueues.forEach(([queueName, queue]) => {
+        queue.process('*', WORKER_CONCURRENCY, async function (job) {
+            logger.info({ taskId: job.id, status: 'processing', queue: queueName, task: getTaskLoggingContext(job) })
+            try {
+                return await executeTask(job.name, job.data.args, job)
+            } catch (error) {
+                logger.error({ taskId: job.id, status: 'error', error, queue: queueName, task: getTaskLoggingContext(job) })
+                throw error
+            }
+        })
+
+        queue.on('failed', function (job) {
+            logger.info({ taskId: job.id, status: 'failed', queue: queueName, task: getTaskLoggingContext(job) })
+        })
+
+        queue.on('completed', function (job) {
+            logger.info({ taskId: job.id, status: 'completed', queue: queueName, task: getTaskLoggingContext(job), t0: job.finishedOn - job.timestamp, t1: job.processedOn - job.timestamp, t2: job.finishedOn - job.processedOn })
```

Comment:
Maybe we can rename t0 and t1, it will not be obvious to read (I guess)

Reply:
done